dask/delayed.py

```diff
@@ -437,7 +461,7 @@ def call_function(func, func_token, args, kwargs, pure=False, nout=None):
 class DelayedLeaf(Delayed):
     __slots__ = ('_obj', '_key', '_pure', '_nout')
 
-    def __init__(self, obj, key, pure=False, nout=None):
+    def __init__(self, obj, key, pure=None, nout=None):
         self._obj = obj
         self._key = key
         self._pure = pure
```

Comment:
This is the same comment as a previous line note. In DelayedAttr below, it seems purity is always assumed True. I'm fine with it if you are okay with it (and @mrocklin ), but maybe it should be documented? (There could be cases like an attribute spitting out a random number. We have image series simulators where indexing an image in the image sequence generates a new random image for example. Of course we could implement this more explicitly for a task graph, so long as we are properly warned about it.)